### PR TITLE
[NEW] Real permissions names and descriptions

### LINKED ISSUE #4371: Permission names to have i18n (real names)
Your Rocket.Chat version: 0.39

Based on the conversation in #1578, especially @geekgonecrazy 's [comment](https://github.com/RocketChat/Rocket.Chat/issues/1578#issuecomment-218847624).

At present, it's really hard to understand what permissions like delete-p actually do.

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core 

Closes #4371

This adds translatable names and descriptions to the permissions

selection_007

also this PR will sort alphabetically  the en.i18n.js file 

*Update*: Display the technical permission name too   
screenshot from 2017-06-13 14-31-59

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
People are going to love this!  :)

**Comment by graywolf336** [comment]
This has been much needed for a long time.

Only issue I have so far is the language of the descriptions of the permissions. `Permission to` seems like it would be implied since they're on the permission screen?

Also, will there be a way for people to see the technical permission anymore? Say they get an error and find the permission name needed but then they come to this screen and can't find it

**Comment by MartinSchoeler** [comment]
@graywolf336 I agree with you, but I'm not sure how this should be written. Any ideas?

**Comment by gdelavald** [comment]
We could add a simple 'Localization String [error-code]' like this error message:
image

**Comment by MartinSchoeler** [comment]
Also if you guys could take a look in the descriptions to see if I got them right. In this commit is easier to read the changes https://github.com/RocketChat/Rocket.Chat/pull/7242/commits/ba357419a4b258a3d788c61bae9162fe425c5ad6

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -1721,8 +1559,314 @@
...
+  "manage-assets": "Manage Assets",
+  "manage-assets_description": "Permission to manage the server assets",
+  "manage-emoji": "Manage Emoji",
+  "manage-emoji_description": "Permission to manage the server emojis",
+  "manage-integrations": "Manage Integrations",
+  "manage-integrations_description": "Permission to manage the server integrations",
+  "manage-oauth-apps": "Manage Oauth Apps",
+  "manage-oauth-apps_description": "Permission to manage the server Oauth apps",
+  "manage-own-integrations": "Manage Own Integrations",
+  "manage-own-integrations_description": "Permition to users create and edit their own integration or webhooks",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: > Permition to users

Should be

> Permission to allow users to

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -1721,8 +1559,314 @@
...
+  "optional": "optional",
+  "or": "or",
+  "others": "others",
+  "pin-message": "Pin Message",
+  "pin-message_description": "Permission to pin a message in a channel",
+  "please_enter_valid_domain": "Please enter a valid domain",
+  "post-readonly": "Post ReadOnly",
+  "post-readonly_description": "Permission to post a message in a read-only channel",
+  "preview-c-room": "Preview Public Channel",
+  "preview-c-room_description": "Permission view the contents of a public channel before joining",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: > Permission view 

should be

> Permission to view

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 3]
```diff
@@ -1721,8 +1559,314 @@
...
+  "theme-color-transparent-lighter": "Transparent Lighter",
+  "theme-color-transparent-lightest": "Transparent Lightest",
+  "theme-color-unread-notification-color": "Unread Notifications Color",
+  "theme-custom-css": "Custom CSS",
+  "theme-font-body-font-family": "Body Font Family",
+  "to_see_more_details_on_how_to_integrate": "to see more details on how to integrate.",
+  "unarchive-room": "Unarchive Room",
+  "unarchive-room_description": "Permission to unarchive channels",
+  "user-generate-access-token": "User Generate Access Token",
+  "user-generate-access-token_description": "Permission to users generate access tokens",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: > Permission to users

Should be

> Permission for users to

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 4]
```diff
@@ -1721,8 +1559,314 @@
...
+  "view-livechat-rooms": "View Livechat Rooms",
+  "view-livechat-rooms_description": "Permission to view other livechat channels",
+  "view-logs": "View Logs",
+  "view-logs_description": "Permission to view the server logs ",
+  "view-other-user-channels": "View Other User Channels",
+  "view-other-user-channels_description": "Permission to view channels owned by other users",
+  "view-p-room": "View Private Room",
+  "view-p-room_description": "Permission to view private channels",
+  "view-privileged-setting": "View Privileged Setting",
+  "view-privileged-setting_description": "Permission to view settings",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: > Permission to view settings

should probably be:

> Permission to view privileged settings

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 5]
```diff
@@ -1721,8 +1559,314 @@
...
+  "view-room-administration": "View Room Administration",
+  "view-room-administration_description": "Permission to view public, private and direct message statistics. Does not include the ability to view conversations or archives",
+  "view-statistics": "View Private Room",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: This should be:

> View Statistics

**Comment by MartinSchoeler** [comment]
Thanks for helping out with the typos @geekgonecrazy

**Comment by geekgonecrazy** [comment]
@MartinSchoeler :+1: with those it looks good to me.  Massive improvement over what we have now, that's for sure!

**Comment by rodrigok** [comment]
@MartinSchoeler Can you fix the conflicts? Probably is better to not sort the lines in your PR

**Pull request merged**